src/mlpack/methods/lmnn/lmnn_function_impl.hpp

```diff
@@ -0,0 +1,501 @@
+/**
+ * @file lmnn_function_impl.cpp
+ * @author Manish Kumar
+ *
+ * An implementation of the LMNNFunction class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_LMNN_FUNCTION_IMPL_HPP
+#define MLPACK_METHODS_LMNN_FUNCTION_IMPL_HPP
+
+#include "lmnn_function.hpp"
+
+#include <mlpack/core/math/make_alias.hpp>
+#include <mlpack/core/optimizers/function.hpp>
+
+namespace mlpack {
+namespace lmnn {
+
+template<typename MetricType>
+LMNNFunction<MetricType>::LMNNFunction(const arma::mat& dataset,
+                           const arma::Row<size_t>& labels,
```

Comment:
Another picky style comment from my side, do you mind to align the method parameter with the first one, if you like you can also put the first parameter on the next line.